### PR TITLE
Add sort=random parameter to the add-ons search API

### LINKED ISSUE #6560: Add random sort ability to Search API
Hi,

In order to retrieve a list of featured add-ons, the addons-frontend project uses the [Featured endpoint](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#featured). Yet, this endpoint does not support other parameters like `appversion`, which we now need to only show compatible add-ons to the users.

The [Search API](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#search) has everything we need to retrieve a similar list of featured add-ons, **except** that we cannot get a random result set (the main point of the Featured endpoint).

By adding a new `sort` type (e.g. `random`), we could rely on the Search API endpoint in the frontend and stop calling the Featured endpoint, which could be then removed.

### PR DESCRIPTION
**Pull request by diox**
Fix #6560

### CODE REVIEW

File: src/olympia/search/filters.py [hunk 1]
```diff
@@ -471,12 +472,38 @@ def filter_queryset(self, request, qs, view):
...
+                raise serializers.ValidationError(
+                    'The "random" "sort" parameter can not be combined.')
+
+            # Second, for perf reasons it's only available when the 'featured'
+            # param is present (to limit the number of documents we'll have to
+            # apply the random score to) and a search query is absent
+            # (to prevent clashing with the score functions coming from a
+            # search query).
+            if sort_param == 'random':
+                is_random_sort_available = (
```
Comment (EnTeQuAk) [inline src/olympia/search/filters.py]: This should be `is_featured_param_available` or am I looking at this the other way around?

File: src/olympia/search/filters.py [hunk 2]
```diff
@@ -471,12 +472,38 @@ def filter_queryset(self, request, qs, view):
...
+                    AddonFeaturedQueryParam.query_param in request.GET and
+                    not search_query_param
+                )
+                if is_random_sort_available:
+                    qs = qs.query(
+                        'function_score', functions=[query.SF('random_score')])
+                else:
+                    raise serializers.ValidationError(
+                        'The "random" "sort" parameter can only be specified '
```
Comment (EnTeQuAk) [inline src/olympia/search/filters.py]: I'd write that a bit different: `The "sort" parameter "random" can only be specified...` but that's just personal preference.

File: src/olympia/search/filters.py (hunk 1 above)
Comment (diox) [reply src/olympia/search/filters.py]: No, I stand by that naming :) The variable is there to check if random sort is available, following the conditions described in the comment above.

Comment (EnTeQuAk) [reply src/olympia/search/filters.py]: hmm, then I was looking at it the other way around, I'm fine with keeping it.

**Pull request merged**